jcapi.go

```diff
@@ -315,3 +353,61 @@ func FindObject(sourceArray []interface{}, fieldName string, compareData interfa
 
 	return OBJECT_NOT_FOUND
 }
+
+func FindObjectByStringRegex(sourceArray []interface{}, fieldName string, regex string) (index int, err error) {
+
+	if len(sourceArray) == 0 {
+		err = fmt.Errorf("Source array is empty. object not found")
+		return
+	}
+
+	//
+	// Get the specified field name of the first struct
+	//
+	s := reflect.ValueOf(sourceArray[0]).FieldByName(fieldName)
+
+	// Make sure the requested field name exists in the struct
+	if s.Kind() == reflect.Invalid {
+		err = fmt.Errorf("Field name specified does not exist within the provided array of structs")
+		return
+	}
+
+	// Make sure the compareData type matches that the field specified by fieldName
+	if s.Type().Kind() != reflect.String {
+		err = fmt.Errorf("Type of field name '%s' must be string", fieldName)
+		return
+	}
+
+	r, err := regexp.Compile(regex)
+	if err != nil {
+		err = fmt.Errorf("Could not compile regex for '%s', err='%s'", regex, err.Error())
+		return
+	}
+
+	//
+	// Walk the array and see if we can find a matching object
+	//
+	for fieldIndex, _ := range sourceArray {
+		s = reflect.ValueOf(sourceArray[fieldIndex]).FieldByName(fieldName)
+
+		// Make sure the requested field name exists in the struct
+		if s.Kind() == reflect.Invalid {
+			err = fmt.Errorf("Field name specified does not exist within the object at array index %d", fieldIndex)
+			return
+		}
+
+		// Make sure the compareData type matches that the field specified by fieldName
+		if s.Type().Kind() != reflect.String {
+			err = fmt.Errorf("Type of field name '%s' in object at array index %d must be string", fieldName, fieldIndex)
+			return
+		}
+
+		if r.Match([]byte(s.String())) {
+			index = fieldIndex
+			return
+		}
+	}
+
+	index = OBJECT_NOT_FOUND
```

Comment:
I thought you had this at the top before?  I like that better, as there are numerous exit paths where err is set, but index will be 0; I think it'd be more robust if index was always == OBJECT_NOT_FOUND when err is not nil.